pkg/sql/opt/xform/testdata/rules/join_order

```diff
@@ -2596,3 +2596,173 @@ project
       │         └── t2.a:5 = 123456 [outer=(5), constraints=(/5: [/123456 - /123456]; tight), fd=()-->(5)]
       └── filters
            └── t1.a:1 = 123456 [outer=(1), constraints=(/1: [/123456 - /123456]; tight), fd=()-->(1)]
+
+# Regression test for #80901. Do not cause a memo cycle when redundant inner
+# join filters can be inferred that reference the right side of a left join, and
+# the inner join is pushed into the left side of the left join.
+
+exec-ddl
+CREATE TABLE table80901_1 (
+  col1_5  DECIMAL,
+  col1_7  BOOL,
+  col1_9  OID,
+  col1_11 STRING,
+  col1_12 STRING,
+  col1_14 STRING,
+  col1_15 STRING,
+  col1_16 STRING,
+  col1_17 STRING
+);
+----
+
+exec-ddl
+CREATE TABLE table80901_3 (col3_2 OID, col3_4 FLOAT8, col3_9 STRING);
+----
+
+memo memo-cycles
```

Comment:
nit: If there is a cycle, the optimizer should panic [here](https://github.com/cockroachdb/cockroach/blob/85b2ebeaec2eac7748c60e8fa2a49822c1973945/pkg/sql/opt/xform/optimizer.go#L471-L479) and opttester should print out the error in the test output. So maybe the `memo-cycles` option is unnecessary?